projects/storefrontlib/src/lib/pwa/pwa.module.ts

```diff
@@ -1,12 +1,35 @@
 import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import {
+  ServiceWorkerModule,
+  ɵangular_packages_service_worker_service_worker_b as RegistrationOptions
+} from '@angular/service-worker';
+
 import { Config, ConfigModule } from '@spartacus/core';
+
 import { defaultPWAModuleConfig, PWAModuleConfig } from './pwa.module-config';
 import { AddToHomeScreenBtnComponent } from './add-to-home-screen-btn/add-to-home-screen-btn.component';
-import { CommonModule } from '@angular/common';
+
+export function pwaConfigurationFactory(
+  pwaConfig: PWAModuleConfig
+): RegistrationOptions {
+  return { enabled: (pwaConfig.production && pwaConfig.pwa.enabled) || false };
```

Comment:
this "|| false" is not necessary.

Reply:
It is in a case when `pwaConfig.production`  is `undefined`